### PR TITLE
Fix onboarding status refacto

### PR DESCRIPTION
**Pull request by martmull**
see comments in https://github.com/twentyhq/twenty/pull/6531#pullrequestreview-2217807413
- rename method
- add check before setting create profile pending user var

### CODE REVIEW

File: packages/twenty-server/src/engine/core-modules/auth/services/sign-in-up.service.ts [hunk 1]
```diff
@@ -181,11 +181,13 @@ export class SignInUpService {
       value: true,
     });
 
-    await this.onboardingService.setOnboardingCreateProfileCompletion({
-      userId: user.id,
-      workspaceId: workspace.id,
-      value: true,
-    });
+    if (firstName === '' && lastName === '') {
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/auth/services/sign-in-up.service.ts:184]: I don't think this is actually right anymore, we should always show the create profile popin, it is prefill with currentInformation

File: packages/twenty-server/src/database/commands/upgrade-version/0-23/0-23-backfill-new-onboarding-user-vars.ts [hunk 2]
```diff
@@ -73,7 +73,7 @@ export class BackfillNewOnboardingUserVarsCommand extends CommandRunner {
       });
 
       for (const user of workspace.users) {
-        await this.onboardingService.setOnboardingCreateProfileCompletion({
+        await this.onboardingService.setOnboardingCreateProfilePending({
```
Comment (charlesBochet) [inline packages/twenty-server/src/database/commands/upgrade-version/0-23/0-23-backfill-new-onboarding-user-vars.ts:76]: right!

File: packages/twenty-server/src/engine/core-modules/auth/services/sign-in-up.service.ts [hunk 3]
```diff
@@ -240,11 +242,13 @@ export class SignInUpService {
...
```
Comment (charlesBochet) [inline packages/twenty-server/src/engine/core-modules/auth/services/sign-in-up.service.ts:245]: same

**Comment by FelixMalfait** [comment]
@martmull @charlesBochet Merging to avoid increasing the stock of PR while many are away. I agree "firstname === ''" doesn't seem very clean but it makes sense to skip profile creation when going through Google. OK for me to merge like as I don't have something better to offer

**Pull request merged**